### PR TITLE
[FLINK-24718][avro] update AVRO dependency to latest version

### PR DESCRIPTION
**Pull request by RyanSkraba**
## What is the purpose of the change

* Update AVRO dependency to the latest version

(Continuing on the work of https://github.com/apache/flink/pull/17623)

## Brief change log

  - Updated org.apache.avro:avro:1.10.0 to org.apache.avro:avro:1.11.0

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is already covered by existing tests, such as *flink-avro/*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: yes
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by RyanSkraba** [comment]
@flinkbot run azure

**Review by MartijnVisser — APPROVED** [review]
It looks good to me and I would like to get it merged, but I'm not sure if there's a preference to wait for the fix for https://issues.apache.org/jira/browse/AVRO-3273

@fapaul What do you think?

**Comment by MartijnVisser** [comment]
@zentol I've discussed this briefly with @fapaul but we would like to get your opinion on it.

**Comment by zentol** [comment]
I'm not really fond of the workaround. Does the avro-maven-plugin actually have to be in sync with the avro version?

**Comment by MartijnVisser** [comment]
@RyanSkraba Can you elaborate on that?

**Comment by MartijnVisser** [comment]
@RyanSkraba Since AVRO 1.11.1 is now released, could you update this PR?

**Comment by RyanSkraba** [comment]
I'd love to see this in 1.16! I'm currently with limited connection, but I'll get this updated to 1.11.1 within the week.

**Comment by RyanSkraba** [comment]
@flinkbot run azure

**Comment by eshadei** [comment]
Hello Team,

Can you please confirm, when we are planning to release Flink 1.6 with Avro 1.11.1?

**Comment by RyanSkraba** [comment]
Hello!  I've rebased, removed the maven workaround that we needed for 1.11.0 and squashed everything together.  There is a CI failure with TPC-DS e2e tests, but I can't seem to determine if it's related!

**Comment by zentol** [comment]
See FLINK-28789 for the cause of the e2e failure.

File: flink-formats/flink-avro/src/test/java/org/apache/flink/formats/avro/typeutils/AvroTypeExtractionTest.java [hunk 1]
```diff
@@ -109,9 +109,9 @@ void testSimpleAvroRead(boolean useMiniCluster, @InjectMiniCluster MiniCluster m
                         + "\"type_nested\": {\"num\": 239, \"street\": \"Baker Street\", \"city\": \"London\", "
                         + "\"state\": \"London\", \"zip\": \"NW1 6XE\"}, "
                         + "\"type_bytes\": \"\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\", "
-                        + "\"type_date\": 2014-03-01, \"type_time_millis\": 12:12:12, \"type_time_micros\": 00:00:00.123456, "
-                        + "\"type_timestamp_millis\": 2014-03-01T12:12:12.321Z, "
-                        + "\"type_timestamp_micros\": 1970-01-01T00:00:00.123456Z, \"type_decimal_bytes\": \"\\u0007Ð\", "
+                        + "\"type_date\": \"2014-03-01\", \"type_time_millis\": \"12:12:12\", \"type_time_micros\": \"00:00:00.123456\", "
```
Comment (zentol) [inline flink-formats/flink-avro/src/test/java/org/apache/flink/formats/avro/typeutils/AvroTypeExtractionTest.java:112]: Can you elaborate on why these quotes were added? Were they rejected by Avro etc.

Comment (RyanSkraba) [reply flink-formats/flink-avro/src/test/java/org/apache/flink/formats/avro/typeutils/AvroTypeExtractionTest.java:112]: Hello!  This was a fix introduced by AVRO-3124, cleaning up the `toString()` representation to look more like valid JSON.

My personal opinion is that nobody should parse a `toString()` return value...  but it has been useful in testing.

**Comment by RyanSkraba** [comment]
Hello! For info, did this make it to the list for 1.16?

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Comment by MartijnVisser** [comment]
> Hello! For info, did this make it to the list for 1.16?

Not yet, but perhaps we can still squeeze it in :)

**Pull request merged**